RoboticonColony/Assets/Code/Roboticon.cs

```
@@ -38,30 +53,94 @@ public Roboticon(Tile selectedTile)
+    /// <param name="completedCustomisations"> List of roboticon customisations already applied to the roboticon</param>
+    /// <param name="prerequisites">List of roboticon customisations required to have already been applied to the roboticon</param>
+    /// <returns> If within the compled customisations list, every item of prerequisites is present, return True, else return False</returns>
+    private bool customisationRequirementsTest(List<RoboticonCustomisation> completedCustomisations, List<RoboticonCustomisation> prerequisites)
     {
-        if (resourceType == ItemType.Roboticon){
-            throw new ArgumentException("Roboticon not valid production type");
+        // If the customisation has Prerequisites then...
+        if (prerequisites.Count > 0)
+        {
+            foreach (RoboticonCustomisation currentCustomisation in prerequisites)
+            {
+                if (!completedCustomisations.Contains(currentCustomisation))
+                {
+                    // if Completed Customisations does not contains one Prerequisites then requirements are not met
+                    return false;
+                }
+            }
+            // if Completed Customisations contains all Prerequisites then requirements are met
+            return true;
         }
-        return bonusProduction[resourceType];
+        // else (Customisation has no Prerequisites)
+        else
+        {
+            // As the customisation requires no prerequisites then the requirements are met
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// Move the Roboticon to the selectedTile
+    /// </summary>
+    /// <param name="selectedTile"> The tile the roboticon should be moved to</param>
+    public void MoveRoboticon(Tile selectedTile)
+    {
+        CurrentTile = selectedTile;
+    }
+
+    /// <summary>
+    /// Remove the Roboticon from the selected tile.
+    /// </summary>
+    public void RemoveRoboticon()

```

This should also set InstalledRoboticon to null on the tile its being removed from.
It should throw an error if it is not on a tile.
Also i think RemoveFromTile is more descriptive because removeroboticon sounds like its being applied on the tile class not the roboticon class

---

Removed completely, Roboticon removal should be handled by tile or tile controller.